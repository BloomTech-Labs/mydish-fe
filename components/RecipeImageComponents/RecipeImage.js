import React, { useState } from 'react';
import { View, ImageBackground } from 'react-native';
import { createPlaceholder } from '../../constants/imagePlaceholders';
import styles from '../../styles/recipeImageStyles';
import CameraIcon from './CameraIcon';

function RecipeImage({ image, setImageModalVisible }) {
  return (
    <View style={styles.imageContainer}>
      <ImageBackground
        style={styles.image}
        source={image ? { uri: image } : createPlaceholder}
      >
        <CameraIcon setImageModalVisible={setImageModalVisible} />
      </ImageBackground>
    </View>
  );
}

export default RecipeImage;
