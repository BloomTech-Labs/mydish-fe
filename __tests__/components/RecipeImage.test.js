import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import RecipeImage from '../../components/RecipeImageComponents/RecipeImage';

describe('<ImageUploadModal />', () => {
  const tree = renderer.create(<RecipeImage />).toJSON();
  it('matches snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
