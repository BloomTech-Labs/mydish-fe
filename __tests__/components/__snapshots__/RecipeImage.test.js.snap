// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageUploadModal /> matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "whitesmoke",
      "flex": 1,
      "height": 300,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "flex-end",
        "height": "100%",
        "justifyContent": "flex-end",
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F1F1F1",
            "borderRadius": 16,
            "height": 32,
            "justifyContent": "center",
            "marginBottom": 16,
            "marginRight": 16,
            "padding": 5,
            "width": 32,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 18,
              "width": 22,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
